### ISSUE TITLE
geobuf does not play well with pbf => 1.3.6

### ISSUE THREAD

**Issue by averas** [mapbox/geobuf#62]
I have a server -> client setup where i encode and transfer GeoJSON using geobuf. This last 24 hours a couple of new releases of pbf has been made (1.3.6 and 2.0.0). When upgrading to any of these geobuf seems to generate (or just parse) malformed GeoJSON.

This is when I encode with 1.3.6 or 2.0.0 on the server side and use a matching pbf version on the client end. If I downgrade to 1.3.5 everything looks fine again.

Not sure if this is a pbf bug or just interplay problems between pbf and geobuf.

**Comment by mourner** [comment]
Oh no! Can you provide a minimal test case that reproduces the bug? cc @RReverser must be something about your encoding changes.

**Comment by mourner** [comment]
All `geobuf` tests pass when updating `pbf` to 1.3.6, so we'll need a specific sample GeoJSON that fails (the smaller the better) to be able to figure out the issue.

**Comment by averas** [comment]
It might be something specific to my deployment. I was a bit opportunistic in posting an issue hoping it might be something obvious that you would immediately catch.. Most of my GeoJSON fails, but not completely, I get weird artefacts such as (green vector line):

1.3.5:
<img width="562" alt="screen shot 2016-05-28 at 15 13 47" src="https://cloud.githubusercontent.com/assets/1205805/15627511/cc0e5938-24e6-11e6-98d5-4b3395f34dee.png">

1.3.6:
<img width="580" alt="screen shot 2016-05-28 at 15 14 19" src="https://cloud.githubusercontent.com/assets/1205805/15627512/d1b04702-24e6-11e6-84ff-08651db5754c.png">

The only difference between the two is the pbf version used for the geobuf transfer.

I'll have a test case (or an excuse) ready for you later today!

**Comment by mourner** [comment]
@averas in addition to a test case, can you test this with `pbf` on commit https://github.com/mapbox/pbf/commit/3ea87dcedb32778de1d4b43b3c43ee972deef761, and the one immediately before that (https://github.com/mapbox/pbf/commit/9ed6e8630e0d829e5f0984f9d8abc0b12655b33e), and confirm whether the second doesn't have the issue whether the first one has?

**Comment by RReverser** [comment]
Yeah, it's weird given that test passes. Can't tell anything for sure until we have a test case.

**Comment by averas** [comment]
Okay guys:

```
$ git clone git@github.com:averas/geobuf-test.git
$ cd geobuf-test
$ npm install
$ node server.js
```

This will install pbf 1.3.6 and use a build of pbf 1.3.6 on the client side. Goto http://localhost:8181/ and watch the malformed geojson.

Then, kill the server and:

```
$ npm install pbf@1.3.5
$ node server.js
```
... and modify public/index.html by hand to change to the 1.3.5 client build of pbf.

Surf to http://localhost:8181/ to see correctly shaped geosjon.

**Comment by mourner** [comment]
@averas is there a way to reproduce this without a server setup, just a sample GeoJSON file that you convert to pbf and back and it becomes malformed?

**Comment by averas** [comment]
Sure. I just wanted to make something that was as close to my own setup as possible to begin with. It seems changing the client side version has no effect though, so it's probably on the encoding side of things.

Give me a minute to create such a case...

**Comment by mourner** [comment]
I got, managed to reproduce with the geojson you had in `server.js`. It's indeed a regression in `pbf` @RReverser.

**Comment by mourner** [comment]
I've confirmed 3ea87dcedb32778de1d4b43b3c43ee972deef761 to be the breaking commit. :(

**Comment by averas** [comment]
And I've confirmed you're faster than me. :-)

**Comment by RReverser** [comment]
Cool, but can I also get a descriptor and the source to see the difference in binary? :smile:

**Comment by averas** [comment]
@RReverser 

https://gist.github.com/averas/48763c776e19b7a4ecdfb700b6bffd49

returns true with 1.3.5 and false with 1.3.6.. you can extract any binary rep. you want from that.

**Comment by mourner** [comment]
Released 1.3.7 and 2.0.1 with the fix. Thanks for catching this bug, I was scared! Thankfully we figure it out on Saturday, before more people hit it in production. :)

**Comment by averas** [comment]
I just reflected over that myself.. less than an hour between provided test case and implemented fix and published releases.. on a Saturday.. Great work!
